### PR TITLE
feat(HLS): Support for HLS key rotation

### LINKED ISSUE #741: HLS key rotation
Related to live support.  We have not tested key rotation in HLS.

### PR DESCRIPTION
**Pull request by varvaruc**
We tested this fix internally our HLS streams but also with the stream mentioned in https://github.com/shaka-project/shaka-player/issues/741#issuecomment-1220036539

test stream: https://content.uplynk.com/playlist/ebbbf2111be144a4971dd8426b70e809.m3u8?rmt=wv&m4fenctype=cenc

license server url: https://content.uplynk.com/wv

Fixes #741

### CODE REVIEW

**Comment by avelad** [comment]
Please, sign the CLA. Thanks!

**Comment by avelad** [comment]
Can you add some tests?

**Comment by avelad** [comment]
Please rebase to resolve the conflicts

File: lib/hls/hls_parser.js
Comment (avelad) [inline lib/hls/hls_parser.js:1926]: Is this comment still valid?

Comment (varvaruc) [reply lib/hls/hls_parser.js:1926]: That should be removed. I'll include this in my next commit, if any other findings.

**Comment by varvaruc** [comment]
Hey guys, anything else I can do for this PR so it gets reviewed / merged ?

**Comment by joeyparrish** [comment]
No, you haven't done anything wrong.  I've been juggling many projects and haven't had time to review yet.  I've also been prioritizing bug fixes over feature review.

But this is at the top of my list now that all the bug fixes have been reviewed.  Thanks!

File: lib/hls/hls_parser.js
Comment (joeyparrish) [inline lib/hls/hls_parser.js:381]: We just added a method to the player interface that may be more appropriate for this.  Please merge main into your branch.  Then instead of filtering the entire manifest on every stream, try this:

```js
  this.playerInterface_.newDrmInfo(stream);
```

File: lib/hls/hls_parser.js
Comment (joeyparrish) [inline lib/hls/hls_parser.js:1926]: Please remove

**Review by joeyparrish — CHANGES_REQUESTED** [review]
Looks great!  Just a few changes I'd like to see.  Thanks!

**Comment by joeyparrish** [comment]
I'm going to make the changes I requested, then merge this PR.

All the content URLs mentioned here and in #741 are unavailable.  So I'm going to take a small risk in the changes that I make to this PR before merging.  I'm changing `filter()` to `newDrmInfo()`, but I will only be unable to test that change with the provided unit test.

**Comment by joeyparrish** [comment]
> I'm going to make the changes I requested, then merge this PR.

Actually, I'm not.  The author has not granted permission for maintainers to make changes.  So I'll follow up with another PR of my own.

**Review by joeyparrish — APPROVED** [review]

**Comment by joeyparrish** [comment]
No problem.  Thanks for contributing!

**Pull request merged**
